File: deps/GraphBLAS/GraphBLAS/@GrB/private/util/gb_is_vector.c
//------------------------------------------------------------------------------
// gb_is_vector: determine if a GrB_matrix is a row or column vector
//------------------------------------------------------------------------------

// SuiteSparse:GraphBLAS, Timothy A. Davis, (c) 2017-2020, All Rights Reserved.
// http://suitesparse.com   See GraphBLAS/Doc/License.txt for license.

//------------------------------------------------------------------------------

#include "gb_matlab.h"

bool gb_is_vector               // true if A is a row or column vector
(
    GrB_Matrix A                // GrB_Matrix to query
)
{

    // check if A is m-by-1 or 1-by-n
    GrB_Index nrows = 0, ncols = 0 ;
    if (A != NULL)
    { 
        OK (GrB_Matrix_nrows (&nrows, A)) ;
        OK (GrB_Matrix_ncols (&ncols, A)) ;
    }
    return (nrows == 1 || ncols == 1) ;
}

File: deps/GraphBLAS/GraphBLAS/@GrB/private/gb_isvector.m
function s = gb_isvector (G)
%GB_ISVECTOR determine if the GraphBLAS matrix is a row or column vector,
% where G is the opaque struct of the GraphBLAS matrix.
% gb_isvector (G) is true for an m-by-n GraphBLAS matrix if m or n is 1.

% SuiteSparse:GraphBLAS, Timothy A. Davis, (c) 2017-2021, All Rights Reserved.
% SPDX-License-Identifier: GPL-3.0-or-later

[m, n] = gbsize (G) ;
s = (m == 1) || (n == 1) ;

